### ISSUE TITLE
Eslint plugin not taking the right tsconfig

### ISSUE THREAD

**Issue by Arjanvanmaren** [microsoft/vscode-eslint#1067]
In our project we have several tsconfig files. 

We have a 
tsconfig.json (for react-native-web)
tsconfig-app.json 
tsconfig-eslint.json

What we are trying to accomplish is to use the tsconfig.eslint.json as the tsconfig for eslint. We have configured it like below.

```
module.exports = {
  parser: '@typescript-eslint/parser', // Specifies the ESLint parser
  parserOptions: {
    project: 'tsconfig-eslint.json',
    tsconfigRootDir: __dirname,
    ecmaVersion: 2020, // Allows for the parsing of modern ECMAScript features
    sourceType: 'module', // Allows for the use of imports
    ecmaFeatures: {
      jsx: true // Allows for the parsing of JSX
    }
  },
  settings: {
    react: {
      version: 'detect' // Tells eslint-plugin-react to automatically detect the version of React to use
    }
  },
  extends: [
    'plugin:react/recommended', // Uses the recommended rules from @eslint-plugin-react
    'plugin:@typescript-eslint/recommended' // Uses the recommended rules from @typescript-eslint/eslint-plugin
  ],
  rules: {
    // Place to specify ESLint rules. Can be used to overwrite rules specified from the extended configs
    // e.g. "@typescript-eslint/explicit-function-return-type": "off",
  }
}
```

After setting this we noticed that the eslint cli takes this configuration into account. But the vscode plugin keeps looking at the tsconfig.json file (not the eslint variant). We found this as changes made in the tsconfig.json does affect the vscode editor linting.

Does anyone has an idea how we should configure the editor or the vscode plugin to look at the right config?

**Comment by dbaeumer** [comment]
@Arjanvanmaren does the validation happen correctly when you use eslint in the terminal. If this is the case can you please provide me with a GitHub repository I can clone that demos what you are experiencing.

**Comment by Arjanvanmaren** [comment]
@dbaeumer Hi, I found out what the issue was.

We had three different tsconfigs as you know:
tsconfig.json (for react-native-web as we can not set a path for a custom tsconfig)
tsconfig-app.json
tsconfig-eslint.json

In the ts-config.json (for react native web) we excluded the paths node as these are not supported by react native web. If we leave these in, the file is edited after running the command ``` react scripts start ```.
```
   "paths": {
      "@domains/*": ["src/domains/*"],
      "@app/*": ["src/app/*"],
      "@shared/*": ["src/shared/*"]
    }
```
The error we got was the following:

```Cannot find module '@app/config/settings' or its corresponding type declarations.ts(2307) ```

This seems to be a typescript error and not a eslint error. So actually I think it is not an issue with the eslint extension but with typescript in combination with vscode. The conclusion is that vscode uses the tsconfig.json as default for typescript checking. There is no way we can set the filename or path for the tsconfig to use for typescript validation in the vscode editor settings like following: 

``` "typescript.tsconfigpath": "C:/Dummy/tsconfig-use-this-one.json" ```

While we can with the cli by doing ```yarn tsc -p tsconfig-eslint.json ```.
